src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -132,11 +132,30 @@ public Test BuildFrom(FixtureMethod method, Test suite)
                     tests.Add(test); 
             }
 
-            return hasBuildersSpecified && method.Method.GetParameters().Length > 0 || tests.Count > 0
+            return hasBuildersSpecified && method.Method.GetParameters().Length > 0 || 
+                   tests.Count > 0 ||
+                   MethodHasDataAttributeParameters(method)
                 ? BuildParameterizedMethodSuite(method, tests)
                 : BuildSingleTestMethod(method, suite);
         }
 
+        /// <summary>
+        /// Checks to see if the method has arguments with DataAttribute, 
+        /// if the argument has no data, the test should not run rather than fail.
+        /// </summary>
+        /// <param name="fixtureMethod">The method for which a test is to be built.</param>
+        /// <returns>True if the method signature contains DataAttribute arguments</returns>
+        private bool MethodHasDataAttributeParameters(FixtureMethod fixtureMethod)
+        {
+            foreach (var parameter in fixtureMethod.Method.GetParameters())
+            {
+                var dataAttributes = parameter.GetAttributes<DataAttribute>(true);
```

Comment:
This should be for `IParameterDataSource` and should use inherited=false because that's what NUnit does in all other places where parameters are used.